backend/pkg/api/handlers/auth.go

```diff
@@ -239,9 +292,95 @@ func (h *HandlerService) InternalPostUserConfirm(w http.ResponseWriter, r *http.
 
 	// TODO: purge all user sessions
 
+	returnNoContent(w)
+}
+
+func (h *HandlerService) InternalPutUserPasswordReset(w http.ResponseWriter, r *http.Request) {
+	var v validationError
+	req := struct {
+		Email string `json:"email"`
+	}{}
+	if err := v.checkBody(&req, r); err != nil {
+		handleErr(w, err)
+		return
+	}
+
+	// validate email
+	email := v.checkEmail(req.Email)
+	if v.hasErrors() {
+		handleErr(w, v)
+		return
+	}
+
+	userId, err := h.dai.GetUserByEmail(r.Context(), email)
+	if err != nil {
+		if err == dataaccess.ErrNotFound {
+			returnConflict(w, errors.New("email not registered"))
+		} else {
+			handleErr(w, err)
+		}
+		return
+	}
+
+	// send password reset email
+	err = h.sendResetEmail(r.Context(), userId, email)
+	if err != nil {
+		handleErr(w, err)
+		return
+	}
+
 	returnOk(w, nil)
 }
 
+func (h *HandlerService) InternalPostUserPasswordResetHash(w http.ResponseWriter, r *http.Request) {
+	var v validationError
+	resetToken := v.checkUserEmailToken(mux.Vars(r)["token"])
```

Comment:
The token only needs to be part of the URL in FE, for the API endpoint we can move this into the request payload.